### PR TITLE
Remove Sphinx as a required dependency

### PR DESCRIPTION
**Pull request by phlax**

### CODE REVIEW

**Comment by andy-maier** [comment]
Gentle reminder to get this in. We are also hampered by the circular dependency (using pipdeptree).

File: CHANGES [hunk 1]
```diff
@@ -1,3 +1,8 @@
+Release 1.0.7 (Pending)
+==========================
```
Comment (AA-Turner) [inline CHANGES:2]:
```suggestion
Release 1.0.7 (unreleased)
==========================
```

File: pyproject.toml [hunk 2]
```diff
@@ -53,7 +50,9 @@ lint = [
     "mypy",
     "docutils-stubs",
 ]
-
+standalone = [
+  "sphinx > 5",
```
Comment (AA-Turner) [inline pyproject.toml:54]:
```suggestion
  "Sphinx>=5",
```

File: CHANGES [hunk 3]
```diff
@@ -1,3 +1,8 @@
+Release 1.0.7 (unreleased)
+==========================
+
+* Fix circular dependency with Sphinx that caused failure in DAG-based package management
```
Comment (AA-Turner) [inline CHANGES:4]:
```suggestion
* Remove Sphinx as a required dependency, as circular dependencies may cause
  failure with package managers that expect a directed acyclic graph (DAG)
  of dependencies.
```

File: pyproject.toml [hunk 4]
```diff
@@ -53,7 +50,9 @@ lint = [
     "mypy",
     "docutils-stubs",
 ]
-
+standalone = [
+  "Sphinx>=5",
```
Comment (AA-Turner) [inline pyproject.toml:54]:
```suggestion
    "Sphinx>=5",
```

File: pyproject.toml [hunk 5]
```diff
@@ -39,9 +39,6 @@ classifiers = [
     "Topic :: Text Processing",
     "Topic :: Utilities",
 ]
-dependencies = [
-    "Sphinx>=5",
-]
 dynamic = ["version"]
```
Comment (AA-Turner) [inline pyproject.toml:42]:
```suggestion
dependencies = []
dynamic = ["version"]
```

File: pyproject.toml [hunk 6]
```diff
@@ -53,7 +51,9 @@ lint = [
     "mypy",
     "docutils-stubs",
 ]
-
+standalone = [
+    "Sphinx>=5",
+]
 [[project.authors]]
```
Comment (AA-Turner) [inline pyproject.toml:57]:
```suggestion

[[project.authors]]
```

**Comment by mvorisek** [comment]
@AA-Turner Please revert, the dep is needed, it has broken our CIs:
```
Run mv docs/baseline.txt docs/baseline.orig.txt
  mv docs/baseline.txt docs/baseline.orig.txt
  (cd docs && python -m sphinx -T -b html . out 2>&1 | tee baseline.txt)
  sed -i -r 's~[^:]*/docs/([^:]*:)([0-9]+:)?~\1~;t;d' docs/baseline.txt
  shell: sh -e {0}
Running Sphinx v4.5.0

Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/sphinx/registry.py", line 438, in load_extension
    metadata = setup(app)
               ^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/sphinxcontrib/applehelp/__init__.py", line 230, in setup
    app.require_sphinx('5.0')
  File "/usr/lib/python3.11/site-packages/sphinx/application.py", line 393, in require_sphinx
    raise VersionRequirementError(version)
sphinx.errors.VersionRequirementError: 5.0

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/sphinx/cmd/build.py", line 272, in build_main
    app = Sphinx(args.sourcedir, args.confdir, args.outputdir,
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/sphinx/application.py", line 219, in __init__
    self.setup_extension(extension)
  File "/usr/lib/python3.11/site-packages/sphinx/application.py", line 380, in setup_extension
    self.registry.load_extension(self, extname)
  File "/usr/lib/python3.11/site-packages/sphinx/registry.py", line 441, in load_extension
    raise VersionRequirementError(
sphinx.errors.VersionRequirementError: The sphinxcontrib.applehelp extension used by this project needs at least Sphinx v5.0; it therefore cannot be built with this version.

Sphinx version error:
The sphinxcontrib.applehelp extension used by this project needs at least Sphinx v5.0; it therefore cannot be built with this version.
```

**Comment by AA-Turner** [comment]
You're using Sphinx v4.5.0, which is no longer supported -- if you require Sphinx 4 still, you'll have to pin dependencies.

A

**Comment by mvorisek** [comment]
@AA-Turner let's do a Sphinx 4.5.1 patch release then fix with https://github.com/sphinx-doc/sphinx/blob/v4.5.0/setup.py#L17 line changed to:
```diff
-    'sphinxcontrib-applehelp',
+    'sphinxcontrib-applehelp<=1.0.4',
```

**Pull request merged**